File: Tests/Test_IsFileOrDirectory.vb
Option Explicit On
Option Strict On
Option Compare Binary
Option Infer Off

Imports System.IO

Namespace Tests
    Module Tests_IsFileOrDirectory
        Function Test_IsFileOrDirectory1(rootTestFolder As String) As Boolean
            Using testDir As DisposableDirectory = New DisposableDirectory(Path.Combine(rootTestFolder, "isFileOrDirectory1"))
                Return TestNumber("IsFileOrDirectory1", WalkmanLib.IsFileOrDirectory(testDir.dirPath), PathEnum.Exists Or PathEnum.IsDirectory)
            End Using
        End Function

        Function Test_IsFileOrDirectory2(rootTestFolder As String) As Boolean
            Using testFile As DisposableFile = New DisposableFile(Path.Combine(rootTestFolder, "isFileOrDirectory2.txt"))
                Return TestNumber("IsFileOrDirectory2", WalkmanLib.IsFileOrDirectory(testFile.filePath), PathEnum.Exists Or PathEnum.IsFile)
            End Using
        End Function

        Function Test_IsFileOrDirectory3() As Boolean
            Return TestNumber("IsFileOrDirectory3", WalkmanLib.IsFileOrDirectory("C:\"), PathEnum.Exists Or PathEnum.IsDirectory Or PathEnum.IsDrive)
        End Function

        Function Test_IsFileOrDirectory4() As Boolean
            Return TestNumber("IsFileOrDirectory4", WalkmanLib.IsFileOrDirectory("C:\nonexistantpath"), PathEnum.NotFound)
        End Function

        Function Test_IsFileOrDirectory5() As Boolean
            Return TestNumber("IsFileOrDirectory5", WalkmanLib.IsFileOrDirectory("test:test\test"), PathEnum.NotFound)
        End Function

        Function Test_IsFileOrDirectory6() As Boolean
            Return TestNumber("IsFileOrDirectory6", WalkmanLib.IsFileOrDirectory("~!@#$%^&*(){}[]/?=+-_\|"), PathEnum.NotFound)
        End Function

        Function Test_IsFileOrDirectory7() As Boolean
            Return TestNumber("IsFileOrDirectory7", WalkmanLib.IsFileOrDirectory("M:\"), PathEnum.Exists Or PathEnum.IsDrive)
        End Function
    End Module
End Namespace

File: Tests/Test_IsFileOrDirectory.cs
using System.IO;

namespace Tests {
    static class Tests_IsFileOrDirectory {
        public static bool Test_IsFileOrDirectory1(string rootTestFolder) {
            using (var testDir = new DisposableDirectory(Path.Combine(rootTestFolder, "isFileOrDirectory1"))) {
                return GeneralFunctions.TestNumber("IsFileOrDirectory1", (int)WalkmanLib.IsFileOrDirectory(testDir), (int)(PathEnum.Exists | PathEnum.IsDirectory));
            }
        }

        public static bool Test_IsFileOrDirectory2(string rootTestFolder) {
            using (var testFile = new DisposableFile(Path.Combine(rootTestFolder, "isFileOrDirectory2.txt"))) {
                return GeneralFunctions.TestNumber("IsFileOrDirectory2", (int)WalkmanLib.IsFileOrDirectory(testFile), (int)(PathEnum.Exists | PathEnum.IsFile));
            }
        }

        public static bool Test_IsFileOrDirectory3() {
            return GeneralFunctions.TestNumber("IsFileOrDirectory3", (int)WalkmanLib.IsFileOrDirectory(@"C:\"), (int)(PathEnum.Exists | PathEnum.IsDirectory | PathEnum.IsDrive));
        }

        public static bool Test_IsFileOrDirectory4() {
            return GeneralFunctions.TestNumber("IsFileOrDirectory4", (int)WalkmanLib.IsFileOrDirectory(@"C:\nonexistantpath"), (int)PathEnum.NotFound);
        }

        public static bool Test_IsFileOrDirectory5() {
            return GeneralFunctions.TestNumber("IsFileOrDirectory5", (int)WalkmanLib.IsFileOrDirectory(@"test:test\test"), (int)PathEnum.NotFound);
        }

        public static bool Test_IsFileOrDirectory6() {
            return GeneralFunctions.TestNumber("IsFileOrDirectory6", (int)WalkmanLib.IsFileOrDirectory(@"~!@#$%^&*(){}[]/?=+-_\|"), (int)PathEnum.NotFound);
        }

        public static bool Test_IsFileOrDirectory7() {
            return GeneralFunctions.TestNumber("IsFileOrDirectory7", (int)WalkmanLib.IsFileOrDirectory(@"M:\"), (int)(PathEnum.Exists | PathEnum.IsDrive));
        }
    }
}
